puppet/lib/puppet/parser/functions/query.rb

```diff
@@ -0,0 +1,45 @@
+Puppet::Parser::Functions.newfunction(:query, :type => :rvalue, :arity => -3, :doc => <<-EOT
+Query PuppetDB
+
+Usage:
+  query(["from", "resources",
+         ["and", ["=", "type", "Class"],
+          ["=", "title", "Puppet_enterprise::Profile::Puppetdb"],
+          ["not", ["=", "certname", $certname]]]])
+  [{"certname" => "foo.bar.com"
+    "environment" => "production",
+    "parameters" => {"master_certname" => "master.foo.lan"
+                     ...}} ...]
+
+  query(["from", "resources",
```

Comment:
Hmm, can this sort of pivot, be outside of the baked in functionality for now? I just feel like a pivot functionality on an array of hashes is fairly generic, and is a modification one can make to the resulting data later on.